python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -177,65 +177,99 @@ def read_partition(
         strings_to_cats = kwargs.get("strings_to_categorical", False)
         read_kwargs = kwargs.get("read", {})
         read_kwargs.update(open_file_options or {})
-
-        # Assume multi-piece read
-        paths = []
-        rgs = []
-        last_partition_keys = None
-        dfs = []
-
-        for i, piece in enumerate(pieces):
-
-            (path, row_group, partition_keys) = piece
-            row_group = None if row_group == [None] else row_group
-
-            if i > 0 and partition_keys != last_partition_keys:
-                dfs.append(
-                    cls._read_paths(
-                        paths,
-                        fs,
-                        columns=read_columns,
-                        row_groups=rgs if rgs else None,
-                        strings_to_categorical=strings_to_cats,
-                        partitions=partitions,
-                        partitioning=partitioning,
-                        partition_keys=last_partition_keys,
-                        **read_kwargs,
+        check_file_size = read_kwargs.pop("check_file_size", None)
+
+        # Wrap reading logic in a `try` block so that we can
+        # inform the user that the `read_parquet` partition
+        # size is too large for the available memory
+        try:
+
+            # Assume multi-piece read
+            paths = []
+            rgs = []
+            last_partition_keys = None
+            dfs = []
+
+            for i, piece in enumerate(pieces):
+
+                (path, row_group, partition_keys) = piece
+                row_group = None if row_group == [None] else row_group
+
+                # File-size check to help "protect" users from change
+                # to up-stream `split_row_groups` default. We only
+                # check the file size if this partition corresponds
+                # to a full file, and `check_file_size` is defined
+                if check_file_size and len(pieces) == 1 and row_group is None:
+                    file_size = fs.size(path)
+                    if file_size > check_file_size:
+                        warnings.warn(
+                            f"A large parquet file ({file_size}B) is being "
+                            f"used to create a DataFrame partition in "
+                            f"read_parquet. Did you mean to use the "
+                            f"split_row_groups argument?"
+                        )
```

Comment:
@randerzander - Does this warning make sense to you?  If the user does not specify `split_row_groups` or `chunksize`, and we end up reading an entire file here, we will warn the user if the file is >2GB on disk.

Note that I also added a `try`/`except` block around the entire data read below to suggest using `split_row_groups` (or to try changing `device_memory_limit`) when we hit a `MemoryError`.

Reply:
> if the file is >2GB on disk.

2GB on disk is likely _huge_ in memory. I often see Parquet files of <100mb  that decompress to 2GB or more.

Reply:
Suggesting to increase device memory limit here feels a bit tangential. I think the `split_row_groups` reference is sufficient.

I think you're highlighting that read_parquet is the "entry point" and thus is likely to see OOMs frequently. Perhaps the docs for read_parquet can cross link to a page about device memory limit?

Reply:
>2GB on disk is likely huge in memory. I often see Parquet files of <100mb that decompress to 2GB or more.

Good point - Do you have a suggestion for a specific number? I want to avoid adding noise for users who really do want a single large file per partition and may have no compression at all (and I'd rather not add the overhead of measuring the partition size in memory after reading)

Reply:
I went with 500MB on disk for now